### PR TITLE
fix(authz): get username from authn.go request context

### PR DESCRIPTION
**Pull request by peusebiu**
**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/api/authz.go [hunk 1]
```diff
@@ -156,19 +156,14 @@ func (ac *AccessController) getUserGroups(username string) []string {
...
 // getContext builds ac context(allowed to read repos and if user is admin) and returns it.
-func (ac *AccessController) getContext(username string, request *http.Request) context.Context {
-	acCtx, err := localCtx.GetAccessControlContext(request.Context())
-	if err != nil {
-		return nil
-	}
-
-	readGlobPatterns := ac.getGlobPatterns(username, acCtx.Groups, Read)
-	dmcGlobPatterns := ac.getGlobPatterns(username, acCtx.Groups, DetectManifestCollision)
+func (ac *AccessController) getContext(acCtx *localCtx.AccessControlContext, request *http.Request) context.Context {
```
Comment (peusebiu) [inline pkg/api/authz.go:159]: rename function to updateContext()

File: pkg/api/authz.go [hunk 2]
```diff
@@ -341,3 +339,13 @@ func anonymousPolicyExists(config *config.AccessControlConfig) bool {
 
 	return false
 }
+
+func isExtensionRequest(request *http.Request) bool {
+	if request.RequestURI == fmt.Sprintf("%s%s", constants.RoutePrefix, constants.ExtCatalogPrefix) ||
```
Comment (andaaron) [inline pkg/api/authz.go:344]: I suggested some changes to this logic in my latest review of https://github.com/project-zot/zot/pull/1317.
But otherwise the proposal looks good to me.

File: pkg/api/authz.go [hunk 3]
```diff
@@ -340,3 +339,13 @@ func anonymousPolicyExists(config *config.AccessControlConfig) bool {
 
 	return false
 }
+
+func isExtensionURI(requestURI string) bool {
```
Comment (andaaron) [inline pkg/api/authz.go:343]: Why this change?

File: pkg/api/authz.go [hunk 4]
```diff
@@ -340,3 +339,13 @@ func anonymousPolicyExists(config *config.AccessControlConfig) bool {
...
+	/* Notes:
+	 	- since we only do READ actions in extensions, we can bypass authz for them
```
Comment (andaaron) [inline pkg/api/authz.go:345]: This note is unrelated to the actual functionality provided by the function. Move it somewhere else?

File: pkg/requestcontext/context.go [hunk 5]
```diff
@@ -43,7 +50,28 @@ func GetAccessControlContext(ctx context.Context) (*AccessControlContext, error)
 	return nil, nil //nolint: nilnil
 }
 
-// returns either a user has or not rights on 'repository'.
+// returns wether or not the user/anonymous who made the request has read permission on 'repository'.
```
Comment (andaaron) [inline pkg/requestcontext/context.go:53]:
```suggestion
// returns whether or not the user/anonymous who made the request has read permission on 'repository'.
```

Same applies to other comments.

File: pkg/api/authz.go (hunk 3 above)
Comment (peusebiu) [reply pkg/api/authz.go]: reverted :D

**Review by andaaron — APPROVED** [review]

**Review by andaaron — DISMISSED** [review]

File: pkg/api/authz.go [hunk 6]
```diff
@@ -155,20 +155,15 @@ func (ac *AccessController) getUserGroups(username string) []string {
...
-// getContext builds ac context(allowed to read repos and if user is admin) and returns it.
-func (ac *AccessController) getContext(username string, request *http.Request) context.Context {
-	acCtx, err := localCtx.GetAccessControlContext(request.Context())
-	if err != nil {
-		return nil
-	}
-
-	readGlobPatterns := ac.getGlobPatterns(username, acCtx.Groups, Read)
-	dmcGlobPatterns := ac.getGlobPatterns(username, acCtx.Groups, DetectManifestCollision)
+// getContext updates an AccessControlContext for a user/anonymous and returns it.
```
Comment (rchincha) [inline pkg/api/authz.go:158]: change "getContext" to "updateContext" in comment

File: pkg/api/authz.go [hunk 7]
```diff
@@ -155,20 +155,15 @@ func (ac *AccessController) getUserGroups(username string) []string {
...
+func (ac *AccessController) updateContext(acCtx *localCtx.AccessControlContext, request *http.Request) context.Context {
```
Comment (peusebiu) [inline pkg/api/authz.go:159]: getContext()
